skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -225,6 +225,9 @@ object RebalanceCallback extends RebalanceCallbackInstances with RebalanceCallba
       }
     }
 
+    @deprecated("For for Scala 3 compact only in places where RebalanceCallback is correctly inferred to `RebalanceCallback[F, ?]` instead of `RebalanceCallback[Nothing, ?]`", since = "16.0.5")
```

Comment:
I think I'm missing some context here. Can you elaborate on this?

Reply:
Without this the `RebalanceListener1Spec` failed at line with `effectAs`, since under Scala 3 the type is correctly inferred to the `RebalanceCallback[IO,Unit]`, but the `effectAs` extension method was only defined for `RebalanceCallback[Nothing, A]` inside `RebalanceCallbackNothingOps`. I thought adding `effectAs` for `RebalanceCallback[F, A]` might help with the migration.

Reply:
We can try to remove that usage, actually. If you specify the return type of `RebalanceListener1Spec.SaveOffsetsOnRebalance#onPartitionsLost` to be `RebalanceCallback[F, Unit]` (it's unspecified now), then you can delete `effectAs` usage and the whole `RebalanceCallbackNothingOps` class, and it seems it compiles just fine.

Reply:
I can remove this line from `RebalanceCallbackOps`, it probably doesn't bring much aside from additional confusion. I'd opt to leave the `RebalanceCallbackNothingOps` as it is though, at least in this PR. There might be usages of it in Scala 2.13 that rely on it.